acorn/src/statement.js

```diff
@@ -617,7 +617,15 @@ pp.parseClassMethod = function(method, isGenerator, isAsync, allowsDirectSuper)
 }
 
 pp.parseClassId = function(node, isStatement) {
-  node.id = this.type === tt.name ? this.parseIdent() : isStatement === true ? this.unexpected() : null
+  if (this.type === tt.name) {
+    node.id = this.parseIdent();
```

Comment:
@TimothyGu Is this `BIND_LEXICAL` specified to be either  `let` or `const`? I ask because the class name should be a `const` inside the class, but a `let` outside of the class, but you seem to use 'lexical'  as a "universal" name.  

See [InitializeBinding(N, V)]( https://tc39.github.io/ecma262/#table-15)

In short it's a `let` binding in outer scope and a `const` binding in inner scope.

Reply:
`BIND_LEXICAL` is considered to be either `let` or `const`, that's correct. However, it doesn't actually matter which one it is at a parser level, as assignment to a `const` is a runtime error, not an early error. In fact, `const a = 2; a = 3;` should (and does) parse.